docs/03_Software/07_FAIR_checklist/checklist.md

```diff
@@ -0,0 +1,94 @@
+
+
+_Recommended_  
+- [ ] Make use of [Github issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues)
+- [ ] Contribution [guidelines](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors)
+- [ ] [Code of conduct](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project)
+
+#### Documentation - Software
+_Essential_  
+- [ ] Source code documentation ([docstrings](https://numpydoc.readthedocs.io/en/latest/format.html))
+- [ ] Installation instructions
+- [ ] Usage documentation
+- [ ] Developer setup
+
+_Recommended_  
+- [ ] Examples and tutorials (Jupyter notebooks, videos, screencasts)
+- [ ] Website ([github.io pages](https://pages.github.com/), [Readthedocs](https://readthedocs.org/), [JupyterBook](https://jupyterbook.org/intro.html)
+- [ ] Build [API reference](https://developer.lsst.io/python/numpydoc.html) from docstrings
+
+
+#### Testing
+_Recommended_  
+- [ ] [Defensive programming](https://swcarpentry.github.io/python-novice-inflammation/10-defensive/index.html)
+- [ ] Integration or build test
+- [ ] User installation test
+- [ ] Unit tests (coverage >70%)
+- [ ] Continuous integration (GitHub Actions)
+- [ ] Code coverage check ([Codecov](https://about.codecov.io/), [Sonarcloud](https://sonarcloud.io/), [Travis](https://www.travis-ci.com/))
+
+
+#### Releases
+_Essential_  
+- [ ] Obtain a DOI ([Zenodo](https://zenodo.org/) or [4TU.ResearchData](https://data.4tu.nl/info/en/))
+
+_Recommended_  
+- [ ] [Semantic versioning](https://semver.org/)
+- [ ] Tagged releases ([GitHub](https://docs.github.com/en/repositories/releasing-projects-on-github))
+- [ ] [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
+- [ ] Upload to registry ([PyPI](https://realpython.com/pypi-publish-python-package/), [conda](https://conda.io/projects/conda-build/en/latest/user-guide/tutorials/build-pkgs.html), [DockerHub](https://docs.docker.com/docker-hub/repos/#:~:text=To%20push%20an%20image%20to,docs%2Fbase%3Atesting%20))
+- [ ] [Release guide](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)
+- [ ] Continuous Integration for build and release
+
+#### Code quality
+_Essential_
+- [ ] Project [organisation](https://coderefinery.github.io/reproducible-research/02-organizing-projects/)
+- [ ] Record [software dependencies](https://coderefinery.github.io/reproducible-research/03-dependencies/)
+
+_Recommended_
+- [ ] Follow [PEP8 guidelines](https://realpython.com/python-pep8/)
+- [ ] Use linter (e.g. [pylint](https://pypi.org/project/pylint/), [flake8](https://pypi.org/project/flake8/))
+- [ ] Use a formatter (e.g. [black](https://github.com/psf/black), [yapf](https://github.com/google/yapf))
+- [ ] Follow [best coding practices](https://alan-turing-institute.github.io/rse-course/html/index.html)
```

Comment:
This is the more precise link with a list of recommendations https://alan-turing-institute.github.io/rse-course/html/module07_construction_and_design/07_03_refactoring.html

Reply:
Good point, but then I would use a link to the entire chapter, as that covers other best-practices as well: https://alan-turing-institute.github.io/rse-course/html/module07_construction_and_design/index.html